modules/caddyhttp/reverseproxy/healthchecks.go

```diff
@@ -153,6 +157,14 @@ func (a *ActiveHealthChecks) Provision(ctx caddy.Context, h *Handler) error {
 	a.httpClient = &http.Client{
 		Timeout:   timeout,
 		Transport: h.Transport,
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			if !a.HealthFollowRedirects {
+				return errors.New(
+					"Redirects are disabled in health check, set health_follow_redirects flag in config to avoid this error")
```

Comment:
We typically prefer `fmt.Errorf` for consistency, avoids an extra import line and makes it easier to add some parameters for formatting if needed.

The newline is not necessary.

Adjusted the message as well.

```suggestion
				return fmt.Errorf("active health check encountered a redirect; enable 'health_follow_redirects' if redirects are intentional")
```

But actually, doing a bit of research, I think it would be better to use `http.ErrUseLastResponse` because that's a special error that the Go stdlib uses as a signal to mean "just return the response as-is without an error if a redirect is encountered". It does mean that users won't see an explanation to use this flag in their logs, but I think that's okay, since we document it elsewhere. They'll instead see an `status code out of tolerances` message in the logs. (/cc @mholt it looks like we actually reject `< 200 || >= 400` which means `3xx` are allowed right now, shouldn't we reduce it to `< 200 || >= 300` ?)

```suggestion
				return http.ErrUseLastResponse
```